src/mapml/layers/MapLayer.js

```diff
@@ -861,31 +861,62 @@ export var MapMLLayer = L.Layer.extend({
                 M.parseStylesheetAsHTML(mapml, base, layer._container);
                 var styleLinks = mapml.querySelectorAll('map-link[rel=style],map-link[rel="self style"],map-link[rel="style self"]');
                 if (styleLinks.length > 1) {
+                  var changeStyle = function (e) {
```

Comment:
Using a mouse pointer it feels a bit unresponsive to not have the style change on click. And using a keyboard it feels a bit disruptive to have the style change happen before intentionally exiting the layer control. But it's certainly an improvement, and fixes the bug. 👍🏼